pymatgen/command_line/bader_caller.py

```diff
@@ -251,28 +276,47 @@ def get_charge_transfer(self, atom_index):
             Given by final charge on atom - nelectrons in POTCAR for
             associated atom.
         """
-        if self.potcar is None:
-            raise ValueError("POTCAR must be supplied in order to calculate "
-                             "charge transfer!")
-        potcar_indices = []
-        for i, v in enumerate(self.natoms):
-            potcar_indices += [i] * v
-        nelect = self.potcar[potcar_indices[atom_index]].nelectrons
-        return self.data[atom_index]["charge"] - nelect
+        if not self.nelects:
+            raise ValueError("No NELECT info! Need POTCAR for VASP, or a structure object"
+                             "with nelect as a site property.")
+        return self.data[atom_index]["charge"] - self.nelects[atom_index]
+
+    def get_charge_decorated_structure(self):
+        """
+        Returns an charge decorated structure
+
+        Note, this assumes that the Bader analysis was correctly performed on a file
+        with electron densities
+
+        """
+        charges = [-self.get_charge(i) for i in range(len(self.structure))]
+        struc = self.structure.copy()
+        struc.add_site_property('charge', charges)
+        return struc
 
     def get_oxidation_state_decorated_structure(self):
         """
-        Returns an oxidation state decorated structure.
+        Returns an oxidation state decorated structure based on bader analysis results.
 
-        Returns:
-            Returns an oxidation state decorated structure. Requires POTCAR
-            to be supplied.
+        Note, this assumes that the Bader analysis was correctly performed on a file
+        with electron densities
+        """
+        charges = [-self.get_charge_transfer(i) for i in range(len(self.structure))]
+        struc = self.structure.copy()
+        struc.add_oxidation_state_by_site(charges)
+        return struc
+
+    def get_spin_state_decorated_structure(self):
+        """
```

Comment:
Is there any way to verify this from the data?

Reply:
Might work. Is there ever a situation where Bader code outputs a negative charge, or is it always positive and you have to calculate the charge transfer? 

If we're confident that it will always be positive, then we can check for negative charge as a filter for spin density files.

Reply:
I don't think there's an unambiguous way to figure out what's in a cube file. Perhaps you could sum up and see if it's the number of electrons you expect (therefore charge) but I'm not sure it's worthwhile.

Reply:
Alright lets skip this idea for now then.